File: app/src/main/java/com/example/pam_project/features/tasks/list/TaskViewHolder.java
package com.example.pam_project.features.tasks.list;

import android.view.View;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.example.pam_project.R;
import com.example.pam_project.features.lists.list.OnListClickedListener;

public abstract class TaskViewHolder extends RecyclerView.ViewHolder {
    private OnListClickedListener listener;

    public TaskViewHolder(@NonNull View itemView) {
        super(itemView);
    }

    public void bind(final TaskInformation task) {
        final TextView title = itemView.findViewById(R.id.title);
        title.setText(task.getTitle());

        itemView.setOnClickListener(v -> {
            if (listener != null) {
                listener.onClick(task.getId());
            }
        });
    }

    public void setOnClickListener(final OnListClickedListener listener) {
        this.listener = listener;
    }
}

File: app/src/main/java/com/example/pam_project/features/tasks/list/TaskViewHolder.kt
package com.example.pam_project.features.tasks.list

import android.view.View
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.example.pam_project.R
import com.example.pam_project.features.lists.list.OnListClickedListener

abstract class TaskViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
    private var listener: OnListClickedListener? = null
    open fun bind(task: TaskInformation?) {
        val title = itemView.findViewById<TextView>(R.id.title)
        title.text = task?.title
        itemView.setOnClickListener {
            if (listener != null) {
                listener!!.onClick(task?.id!!)
            }
        }
    }

    fun setOnClickListener(listener: OnListClickedListener?) {
        this.listener = listener
    }
}